### PR TITLE
Add unit tests

### LINKED ISSUE #64: Unit tests of distribution calculations
To add unit tests that checks that Paymaster distribute correct amount of tokens in non trivial situations.

### LINKED ISSUE #70: Necessary node history is deleted during clearHistory call
**Describe the bug**
Necessary node history is deleted during clearHistory call

**To Reproduce**
To the particular validator
1. set active nodes amount to 1
2. skip time or wait any amount of time
3. let the current time is `middle_time`
4. skip time or wait any amount of time
5. let the current time is `finish_time`
6. set active nodes amount to 2
7. call `clearHistory(middle_time)`

**Expected behavior**
Node history is deleted till `middle_time`

** Actual behavior**
Node history is deleted till `finish_time`

### PR DESCRIPTION
**Pull request by DimaStebaev**
- fix bugs
  - don't delete required nodes history #70 
  - fix rounding error when schain is paid too late
  - revert on payment for 0 months
  - https://github.com/skalenetwork/internal-support/issues/930
- add tests
  - should not add validator with the same id
  - should not add validator with the same address
  - should not allow to top up more than max replenishment period
  - should set version
  - should allow to blacklist nodes
  - should allow to get information about validators and schains
  - should allow admin to claim rewards for the validator
  - should revert on incorrect time interval
  - should not allow to clean unprocessed data
  - should not allow to add to the processed part
  - should get value
  - should get value of empty sequence
  - should clear values
  - big random test
- improve tests
  - should remove validator
- refactor
  - remove long lines

Resolves #64 
Resolves #70

### CODE REVIEW

File: test/tools/matchers.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+            // eslint-disable-next-line no-invalid-this
+            this.catch = derivedPromise.catch.bind(derivedPromise);
+
+            // eslint-disable-next-line no-invalid-this
+            return this;
+        }
+    );
+}
+
+export const d2ChaiMatchers = (
```
Comment (payvint) [inline test/tools/matchers.ts:128]: Easter egg?)

File: contracts/Paymaster.sol [hunk 2]
```diff
@@ -637,7 +697,11 @@ contract Paymaster is AccessManagedUpgradeable, IPaymaster {
         );
 
         DebtId end = debtsEnd;
-        for (DebtId debtId = validator.firstUnpaidDebt; _before(debtId, end); debtId = _next(debtId)) {
+        for (
+            DebtId debtId = validator.firstUnpaidDebt;
```
Comment (payvint) [inline contracts/Paymaster.sol:701]: Impressed!

**Review by payvint — APPROVED** [review]
Nice!!!

File: test/tools/matchers.ts (hunk 1 above)
Comment (DimaStebaev) [reply test/tools/matchers.ts:128]: An author signature.

**Pull request merged**